apps/api/src/domain/utilities/buildEmptyInMemorySnapshot.ts

```diff
@@ -0,0 +1,14 @@
+import { InMemorySnapshot, InMemorySnapshotOfResources, ResourceType } from '../types/ResourceType';
+
+export default (): InMemorySnapshot => ({
+    resources: Object.values(ResourceType).reduce(
+        (accumulatedSnaphshot: InMemorySnapshotOfResources, resourceType) => ({
+            ...accumulatedSnaphshot,
+            [resourceType]: [],
+        }),
+        {}
+    ),
+    tags: [],
```

Comment:
We need to grab these keys from a constant \ enum. Also, we need to standardize whether they should be plural or not.